### PR TITLE
Pixelflat

### PR DESCRIPTION
**Pull request by rcooke-ast**
Added bspline model fit to the flat field (and now takes into account the tilts). This should now be used as the default normalization method.

### CODE REVIEW

File: pypit/arparse.py [hunk 1]
```diff
@@ -992,6 +992,7 @@ def reduce_flatfield_params(self, v):
         specified by the 'reduce flatfield method' keyword:
 
         polyscan:  [Polynomial order, Number of pixels, Number of repeats]
+        bspline:   [Number of pixels in the dispersion direction between each knot]
```
Comment (profxj) [inline pypit/arparse.py]: Maybe add '2D' i.e.   Number of 2D pixels in the dispersion..

File: pypit/arproc.py [hunk 2]
```diff
@@ -586,6 +581,26 @@ def flatnorm(slf, det, msflat, maskval=-999999.9, overpix=6, plotdesc=""):
...
+                                    everyn=slf._pixwid[det-1][o]*settings.argflag['reduce']['flatfield']['params'][0])
+            model_flat = arutils.func_val(bspl, tilts.flatten(), 'bspline')
+            model = model_flat.reshape(tilts.shape)
+            word = np.where(ordpix == o+1)
+            msnormflat[word] = msflat[word]/model[word]
+            msblaze[:, o] = arutils.func_val(bspl, np.linspace(0.0, 1.0, msflat.shape[0]), 'bspline')
+            mskord[word] = 1.0
+            flat_ext1d[:, o] = np.sum(msflat * mskord, axis=1)/np.sum(mskord, axis=1)
+            mskord *= 0.0
+            # import astropy.io.fits as pyfits
```
Comment (profxj) [inline pypit/arproc.py]: Remove or wrap into a   if debug:   statement

Comment (rcooke-ast) [inline pypit/arproc.py]: Removed  :-)

File: pypit/arparse.py (hunk 1 above)
Comment (rcooke-ast) [inline pypit/arparse.py]: This is in fact 1D, because I multiply this argument by the length of the slit (in pixels).

**Comment by rcooke-ast** [comment]
By the way, thanks for reminding me about the docs! I've now updated flatfielding.rst

**Comment by profxj** [comment]
Merging.

**Pull request merged**